element-template-generator/core/src/main/java/io/camunda/connector/generator/java/util/OperationBasedConnectorUtil.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.generator.java.util;
+
+import io.camunda.connector.api.annotation.Operation;
+import io.camunda.connector.api.annotation.Variable;
+import io.camunda.connector.generator.dsl.*;
+import java.lang.reflect.Parameter;
+import java.util.List;
+
+public class OperationBasedConnectorUtil {
+
+  public static String OPERATION_PROPERTY_ID = "operation";
+  public static String OPERATION_TASK_HEADER_KEY = OPERATION_PROPERTY_ID;
+  public static String OPERATION_PROPERTY_SEPARATOR = ":";
+  public static String VARIABLE_PATH_SEPARATOR = ".";
+
+  public static PropertyBuilder createOperationsDropdown(
+      List<ReflectionUtil.MethodWithAnnotation<Operation>> methods) {
+    return new DropdownProperty.DropdownPropertyBuilder()
+        .choices(
+            methods.stream()
+                .map(
+                    m -> {
+                      Operation operation = m.annotation();
+                      String operationName = getOperationName(operation);
+                      String operationId = getOperationId(operation);
+                      return new DropdownProperty.DropdownChoice(operationName, operationId);
+                    })
+                .toList())
+        .id(OPERATION_PROPERTY_ID)
+        .binding(new PropertyBinding.ZeebeTaskHeader(OPERATION_TASK_HEADER_KEY))
+        .label("Operation")
+        .description("The operation to execute")
+        .feel(Property.FeelMode.disabled)
+        .group("operation");
+  }
+
+  public static String getOperationName(Operation operation) {
+    return !operation.name().isBlank() ? operation.name() : operation.value();
+  }
+
+  public static String getOperationId(Operation operation) {
+    return !operation.id().isBlank() ? operation.id() : operation.value();
+  }
```

Comment:
this can still return an empty string if nothing is set, maybe we can validate it here